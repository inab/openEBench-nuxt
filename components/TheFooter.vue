<template>
	<v-main tag="footer" class="footer primary white--text">
		<v-row>
			<v-col class="pa-4" md="3" cols="12" align="center">
				<v-img :src="logo" alt="logo" max-width="150" contain />
				<p class="body-2">&copy;&nbsp; BSC-CNS 2022</p>
			</v-col>
			<v-col class="pa-4" md="6" cols="12">
				<v-expansion-panels v-if="isMobile">
					<v-expansion-panel
						v-for="(footer, index) in footers"
						:key="index"
						class="accordion-content"
					>
						<v-expansion-panel-header>
							<h6 class="title mb-4">
								{{ footer.headline }}
							</h6>
						</v-expansion-panel-header>
						<v-expansion-panel-content>
							<ul>
								<li v-for="(link, n) in footer.links" :key="n">
									<nuxt-link v-if="link.to" :to="link.to">
										{{ link.title }}
									</nuxt-link>
									<a v-if="link.href" :href="link.href" target="_blank">
										{{ link.title }}
									</a>
									<v-icon v-if="link.href" small right>mdi-open-in-new</v-icon>
								</li>
							</ul>
						</v-expansion-panel-content>
					</v-expansion-panel>
				</v-expansion-panels>
				<v-row v-if="isDesktop" justify="space-around">
					<v-col
						v-for="(footer, index) in footers"
						:key="index"
						class="pa-4 site-map-item"
					>
						<h6 class="title mb-4">
							{{ footer.headline }}
						</h6>
						<ul>
							<li v-for="(link, m) in footer.links" :key="m">
								<nuxt-link
									v-if="link.to"
									dark
									class="white--text"
									:to="link.to"
								>
									{{ link.title }}
								</nuxt-link>
								<a
									v-if="link.href"
									:href="link.href"
									target="_blank"
									dark
									class="white--text"
								>
									{{ link.title }}
								</a>
								<v-icon v-if="link.href" small right dark
									>mdi-open-in-new</v-icon
								>
							</li>
						</ul>
					</v-col>
				</v-row>
			</v-col>
			<v-col md="3" cols="12" class="pa-4">
				<div>
					<v-btn
						text
						icon
						dark
						class="button"
						href="mailto:openebench-support@bsc.es"
						aria-label="Mail link for OpenEBench Support"
					>
						<v-icon>mdi-email-outline</v-icon>
					</v-btn>
					<v-btn
						text
						icon
						dark
						class="button"
						href="https://twitter.com/openebench"
						target="_blank"
						aria-label="Link to Twitter Profile of OpenEBench"
					>
						<v-icon>mdi-twitter</v-icon>
					</v-btn>
				</div>
				<v-row no-gutters class="mt-5">
					<v-col cols="9" align="left">
						<p class="text-sm-body-2">
							This project receives funding from the @EU_H2020 Research &
							Innovation programme.
						</p>
					</v-col>
					<v-col cols="3" align="center">
						<v-img :src="euFlag" alt="logo" max-width="70" contain />
					</v-col>
				</v-row>
			</v-col>
		</v-row>
	</v-main>
</template>

<script>
import logo from '~/static/images/opeb_logo_white_minimal.png';
import euFlag from '~/static/images/eu.svg';
import menuEntries from '~/components/Header/menuEntries';
import subMenuEntries from '~/components/Header/subMenuEntries';

export default {
	data: () => ({
		logo,
		euFlag,
		lang: 'en',
		footers: [
			{
				headline: 'OpenEBench',
				links: menuEntries,
			},
			{
				headline: 'Project',
				links: subMenuEntries,
			},
			{
				headline: 'Legal',
				links: [
					{ title: 'Legal Notice', to: '/legal-notice' },
					{ title: 'Disclaimer', to: '/disclaimer' },
					{ title: 'Cookies Policy', to: '/cookies-policy' },
				],
			},
		],
	}),
	computed: {
		isMobile() {
			return this.$vuetify.breakpoint.smAndDown;
		},
		isDesktop() {
			return this.$vuetify.breakpoint.mdAndUp;
		},
	},
};
</script>

<style lang="scss">
.Cookie--dark-lime .Cookie__button {
	background: var(--v-info-base);
}
.Cookie--dark-lime .Cookie__button:hover {
	background: var(--v-info-lighten1);
}
</style>
