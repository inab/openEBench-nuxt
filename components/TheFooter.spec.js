import { mount } from '@vue/test-utils';
import TheFooter from './TheFooter.vue';

const factory = () => {
	return mount(TheFooter, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
	});
};

describe('TheFooter', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should have computed prop isMobile true on small devices', async () => {
		const wrapper = factory();

		wrapper.vm.$vuetify.breakpoint.smAndDown = true;
		wrapper.vm.$vuetify.breakpoint.mdAndUp = false;

		await wrapper.vm.$nextTick();
		expect(wrapper.vm.isMobile).toBe(true);
		expect(wrapper.vm.isDesktop).toBe(false);
	});

	it('should have computed prop isDesktop true on small devices', async () => {
		const wrapper = factory();

		wrapper.vm.$vuetify.breakpoint.smAndDown = false;
		wrapper.vm.$vuetify.breakpoint.mdAndUp = true;

		await wrapper.vm.$nextTick();
		expect(wrapper.vm.isDesktop).toBe(true);
		expect(wrapper.vm.isMobile).toBe(false);
	});
});
