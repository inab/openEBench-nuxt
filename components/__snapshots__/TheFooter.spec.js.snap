// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheFooter should match snapshot 1`] = `
"<footer class=\\"v-main footer primary white--text\\" style=\\"padding: 0px 0px 0px 0px;\\">
  <div class=\\"v-main__wrap\\">
    <div class=\\"row\\">
      <div align=\\"center\\" class=\\"pa-4 col-md-3 col-12\\">
        <div aria-label=\\"logo\\" role=\\"img\\" class=\\"v-image v-responsive theme--light\\" style=\\"max-width: 150px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <p class=\\"body-2\\">Â©&nbsp; BSC-CNS 2022</p>
      </div>
      <div class=\\"pa-4 col-md-6 col-12\\">
        <!---->
        <div class=\\"row justify-space-around\\">
          <div class=\\"pa-4 site-map-item col\\">
            <h6 class=\\"title mb-4\\">
              OpenEBench
            </h6>
            <ul>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Benchmarking
                </a>
                <!---->
                <!---->
              </li>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Observatory
                </a>
                <!---->
                <!---->
              </li>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Projects
                </a>
                <!---->
                <!---->
              </li>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Tools
                </a>
                <!---->
                <!---->
              </li>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Stats
                </a>
                <!---->
                <!---->
              </li>
            </ul>
          </div>
          <div class=\\"pa-4 site-map-item col\\">
            <h6 class=\\"title mb-4\\">
              Project
            </h6>
            <ul>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  About
                </a>
                <!---->
                <!---->
              </li>
              <li>
                <!----> <a href=\\"https://docs.openebench.bsc.es/\\" target=\\"_blank\\" dark=\\"\\" class=\\"white--text\\">
                  Github Docs
                </a> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-icon--right mdi mdi-open-in-new theme--dark\\" style=\\"font-size: 16px;\\"></i></li>
              <li>
                <!----> <a href=\\"https://openebench.readthedocs.io/en/latest/\\" target=\\"_blank\\" dark=\\"\\" class=\\"white--text\\">
                  Project Docs
                </a> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-icon--right mdi mdi-open-in-new theme--dark\\" style=\\"font-size: 16px;\\"></i></li>
            </ul>
          </div>
          <div class=\\"pa-4 site-map-item col\\">
            <h6 class=\\"title mb-4\\">
              Legal
            </h6>
            <ul>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Legal Notice
                </a>
                <!---->
                <!---->
              </li>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Disclaimer
                </a>
                <!---->
                <!---->
              </li>
              <li><a dark=\\"\\" class=\\"white--text\\">
                  Cookies Policy
                </a>
                <!---->
                <!---->
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div class=\\"pa-4 col-md-3 col-12\\">
        <div><a href=\\"mailto:openebench-support@bsc.es\\" class=\\"button v-btn v-btn--icon v-btn--round v-btn--text theme--dark v-size--default\\" aria-label=\\"Mail link for OpenEBench Support\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-email-outline theme--dark\\"></i></span></a> <a href=\\"https://twitter.com/openebench\\" target=\\"_blank\\" class=\\"button v-btn v-btn--icon v-btn--round v-btn--text theme--dark v-size--default\\" aria-label=\\"Link to Twitter Profile of OpenEBench\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-twitter theme--dark\\"></i></span></a></div>
        <div class=\\"row mt-5 no-gutters\\">
          <div align=\\"left\\" class=\\"col col-9\\">
            <p class=\\"text-sm-body-2\\">
              This project receives funding from the @EU_H2020 Research &amp;
              Innovation programme.
            </p>
          </div>
          <div align=\\"center\\" class=\\"col col-3\\">
            <div aria-label=\\"logo\\" role=\\"img\\" class=\\"v-image v-responsive theme--light\\" style=\\"max-width: 70px;\\">
              <div class=\\"v-responsive__content\\"></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</footer>"
`;
